### PR TITLE
feat: issue #9282 not creating page_pruning_predicate when pruning is…

### PR DESCRIPTION
**Pull request by Lordworms**
… diable

## Which issue does this PR close?
#9282 

Closes #.

### CODE REVIEW

File: datafusion/core/src/datasource/physical_plan/parquet/mod.rs [hunk 1]
```diff
@@ -120,33 +121,41 @@ impl ParquetExec {
             MetricBuilder::new(&metrics).global_counter("num_predicate_creation_errors");
 
         let file_schema = &base_config.file_schema;
-        let pruning_predicate = predicate
-            .clone()
-            .and_then(|predicate_expr| {
-                match PruningPredicate::try_new(predicate_expr, file_schema.clone()) {
+        let pruning_predicate = if config.pruning {
```
Comment (SteveLauC) [inline datafusion/core/src/datasource/physical_plan/parquet/mod.rs:124]: It seems that whether this pushdown is enabled is decided by `ParquetFormat.enable_pruning || config.pruning`.

`ParquetFormat.enable_pruning` is used in `ParquetFormat::create_physical_plan`.

Comment (Lordworms) [reply datafusion/core/src/datasource/physical_plan/parquet/mod.rs:124]: I would add that

**Review by Jefffrey — COMMENTED** [review]
Could you help me to understand the impact of this change? From what I can tell, `ParquetFormat::create_physical_plan()` already checks this config before passing in the optional predicate, so this check seems redundant? Unless the intention is to centralize this check within `ParquetExec::new()` itself?

**Comment by Lordworms** [comment]
> Could you help me to understand the impact of this change? From what I can tell, `ParquetFormat::create_physical_plan()` already checks this config before passing in the optional predicate, so this check seems redundant? Unless the intention is to centralize this check within `ParquetExec::new()` itself?

Yes, what I get from this issue is that @SteveLauC  wants to centralize this check with ParquetExec::new()

**Comment by SteveLauC** [comment]
Well, centralizing the configuration to ParquetExec is not what I want, I simply don't want these 2 predicates to be created if the pushdowns are disabled.


Actually, does the issue described in #9282 really exist? I just gave another look at the calling procedure, it seems that if `prune` is disabled, then the `predicate` argument of `ParquetExec::new()` will be None, then these 2 predicates won't be created.

**Comment by SteveLauC** [comment]
It seems that if `prune` is disabled, then filter pushdown will also be disabled as the `predicate` argument of `ParquetExec::new()` will be `None`.

**Comment by Lordworms** [comment]
> It seems that if `prune` is disabled, then filter pushdown will also be disabled as the `predicate` argument of `ParquetExec::new()` will be `None`.

I know that. But I thought you want to centralized the check in ParquestExec::new(). lol

**Comment by SteveLauC** [comment]
For this PR and #9282, I am sorry that I opened an issue that actually does not exist:D

-----

> Off-topic warning: the below behavior is not related to this PR

> It seems that if prune is disabled, then filter pushdown will also be disabled as the predicate argument of ParquetExec::new() will be None.

I am curious if this behavior is wanted?

**Comment by Jefffrey** [comment]
> Off-topic warning: the below behavior is not related to this PR
>
> It seems that if prune is disabled, then filter pushdown will also be disabled as the predicate argument of ParquetExec::new() will be None.
>
> I am curious if this behavior is wanted?

This might be worth opening a new question issue for some more discussion on it? :thinking:

**Comment by alamb** [comment]
I am a little confused about what the status of this PR is -- is it still something we want to consider or have we decided it is not needed because https://github.com/apache/arrow-datafusion/issues/9282 was not a valid request?

**Comment by Lordworms** [comment]
Yes, I think it is not valid, I would close it

**Pull request closed without merging**